kernel/src/hil/debug.rs

```diff
@@ -0,0 +1,15 @@
+/// Interface for interacting with debug hardware integrated in various SoCs.
```

Comment:
"debug.rs" is somewhat generic, and there is the potential for confusion with `kernel/src/debug.rs`. Is there a name that would be somewhat more informative? Maybe `hw_debug`?